src/partition.ml

```diff
@@ -0,0 +1,120 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
+type entry = { name : string; kind : [ `Node | `Contents ]; hash : Hash.t }
+[@@deriving irmin]
```

Comment:
why do you need the irmin derivers for these types?

Reply:
I needed it to debug but it was not meant to stay and has been removed now ;-)